<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Open Software for Teaching Evolutionary Biology at Multiple Scales Through Virtual Inquiries • evolved</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Open Software for Teaching Evolutionary Biology at Multiple Scales Through Virtual Inquiries">
<meta property="og:description" content="&quot;Evolutionary Virtual Education&quot; - evolved - provides multiple tools to help educators (especially at the graduate level or in advanced undergraduate level courses) apply inquiry-based learning in general evolution classes. In particular, the tools provided include functions that simulate evolutionary processes (e.g., genetic drift, natural selection within a single locus) or concepts (e.g. Hardy-Weinberg equilibrium, phylogenetic distribution of traits). More than only simulating, the package also provides tools for students to analyze (e.g., measuring, testing, visualizing) datasets with characteristics that are common to many fields related to evolutionary biology. Importantly, the package is heavily oriented towards providing tools for inquiry-based learning - where students follow scientific practices to actively construct knowledge. For additional details, see package's vignettes.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">evolved</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/birthdeath_deeptime.html">Birth-death models and deep time</a>
    </li>
    <li>
      <a href="articles/birthdeath_phylogenies.html">Birth-death models and molecular phylogenies</a>
    </li>
    <li>
      <a href="articles/deeptime_clocks.html">Allele frequency change and selection</a>
    </li>
    <li>
      <a href="articles/deeptime_rocks.html">Introduction to the fossil record</a>
    </li>
    <li>
      <a href="articles/intro_R.html">Introduction to R</a>
    </li>
    <li>
      <a href="articles/phylogenies_and_diversification.html">Phylogenies and diversification</a>
    </li>
    <li>
      <a href="articles/popgen_drift.html">Sampling alleles, genetic drift and effective population size</a>
    </li>
    <li>
      <a href="articles/popgen_intro.html">Introduction to evolutionary genetics</a>
    </li>
    <li>
      <a href="articles/popgen_selection.html">Allele frequency change and selection</a>
    </li>
    <li>
      <a href="articles/population_genetics.html">Tools for population genetics</a>
    </li>
    <li>
      <a href="articles/rocks_and_clocks.html">Clocks and Rocks</a>
    </li>
  </ul>
</li>
<li>
  <a href="news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/mjanuario/evolved/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><div class="row">
  <div class="contents col-md-9">

<div class="section level1">
<div class="page-header"><h1 id="instructions-on-evolved-r-rstudio-rmarkdown">Instructions on evolved, R, RStudio, RMarkdown<a class="anchor" aria-label="anchor" href="#instructions-on-evolved-r-rstudio-rmarkdown"></a>
</h1></div>
<p>Matheus Januario, Jenniefer Auler, Andressa Viol, and Daniel Rabosky Nov/2024</p>
<ul>
<li>
<a href="#evolved">evolved</a>
<ul>
<li><a href="#installing-evolved">Installing evolved</a></li>
<li><a href="#vignettes-tutorials">Vignettes (Tutorials)</a></li>
</ul>
</li>
<li>
<a href="#if-you-never-used-r-before">If you never used R before…</a>
<ul>
<li><a href="#what-is-r-and-how-can-i-install-it">What is R and how can I install it?</a></li>
<li><a href="#what-is-rstudio-and-how-can-i-install-it">What is RStudio and how can I install it?</a></li>
<li><a href="#starting-r-and-rstudio">Starting R and RStudio:</a></li>
<li><a href="#what-is-rmarkdown-and-how-can-i-install-it-optional">What is RMarkdown and how can I install it (optional)?</a></li>
<li><a href="#working-routine-in-r">Working Routine in R</a></li>
</ul>
</li>
</ul>
</div>
<div class="section level1">
<h1 id="evolved">evolved<a class="anchor" aria-label="anchor" href="#evolved"></a>
</h1>
<p><code>evolved</code> (EVOLutionary Virtual EDucation) is an open-source R package designed for graduate or advanced undergraduate courses in evolutionary biology. It emphasizes tools for inquiry-based learning, where students engage in scientific practices to actively build knowledge (Pedaste et al., 2015). The package includes <code>vignettes</code> (tutorials) to facilitate classroom investigations. However, educators are encouraged to develop their own content modules depending on class context and/or learning objectives. Most of evolved‘s core functions are oriented towards either (<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>i</mi><annotation encoding="application/x-tex">i</annotation></semantics></math>) developing student intuition for classic models in evolutionary biology using simulations, or (<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>i</mi><mi>i</mi></mrow><annotation encoding="application/x-tex">ii</annotation></semantics></math>) analyzing (parameter estimation, hypothesis testing, visualization) datasets associated with key questions in evolutionary biology.</p>
<div class="section level2">
<h2 id="installing-evolved">Installing evolved<a class="anchor" aria-label="anchor" href="#installing-evolved"></a>
</h2>
<p>The evolved package is available in two versions:</p>
<div class="section level3">
<h3 id="stable-version">Stable Version:<a class="anchor" aria-label="anchor" href="#stable-version"></a>
</h3>
<p>The stable version represents the “error-proof” version of the package. It may have reduced features due to CRAN’s constraints (e.g., dataset size). The stable version can be downloaded directly from the <a href="https://cran.r-project.org/web/packages/evolved/index.html" class="external-link">The Comprehensive R Archive Network</a>. A simpler way to install this version is to run the following command in your R terminal:</p>
<pre><code><span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"evolved"</span><span class="op">)</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="development-version">Development Version:<a class="anchor" aria-label="anchor" href="#development-version"></a>
</h3>
<p>The development version is the most complete and updated version of the package. It may contain datasets, functions, or documentation slightly out of CRAN’s standards. The development version is hosted on <a href="https://github.com/mjanuario/evolved" class="external-link">GitHub</a> and requires the <code>devtools</code> package for installation. If <code>devtools</code> is not already installed, you can install it first by running the code lines below:</p>
<pre><code><span><span class="co"># Install devtools</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"devtools"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Then, install the development version of evolved:</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"mjanuario/evolved"</span><span class="op">)</span></span></code></pre>
<div class="section level4">
<h4 id="features-in-the-development-version-that-are-absent-in-the-stable-version">Features in the development version that are absent in the stable version:<a class="anchor" aria-label="anchor" href="#features-in-the-development-version-that-are-absent-in-the-stable-version"></a>
</h4>
<ul>
<li>Datasets:
<ul>
<li>
<code>mammals_spp</code> (Extant species list of mammals)</li>
<li>
<code>ammonoidea_fossil</code> (Fossil occurrences of ammonoids</li>
<li>
<code>mammals_fossil</code> (Fossil occurrences of mammals)</li>
<li>
<code>trilob_fossil</code> (Fossil occurrences of trilobites)</li>
</ul>
</li>
</ul>
</div>
</div>
</div>
<div class="section level2">
<h2 id="vignettes-tutorials">Vignettes (Tutorials)<a class="anchor" aria-label="anchor" href="#vignettes-tutorials"></a>
</h2>
<p>To view the vignettes, run the following code:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="fu">vignette</span>(<span class="st">"vignette_name"</span>, <span class="at">package =</span> <span class="st">"evolved"</span>)</span></code></pre></div>
<p>With <code>"vignette_name"</code> being one of the names below. Each vignette covers the following topics, organized from basic to advanced:</p>
<div class="section level4">
<h4 id="id_1-intro-to-r-name-intro_r">1. Intro to R (name: <code>intro_r</code>)<a class="anchor" aria-label="anchor" href="#id_1-intro-to-r-name-intro_r"></a>
</h4>
<ul>
<li>Basic R syntax and coding (objects, vector calculations, etc.)</li>
<li>Plotting and annotation functions</li>
<li>Overview of key object classes used in the vignettes</li>
</ul>
</div>
<div class="section level4">
<h4 id="id_2-introduction-to-population-genetics-name-popgen_intro">2. Introduction to Population Genetics (name: <code>popgen_intro</code>)<a class="anchor" aria-label="anchor" href="#id_2-introduction-to-population-genetics-name-popgen_intro"></a>
</h4>
<ul>
<li>Simple mathematical notation</li>
<li>Probability of independent events and random number generation</li>
<li>Malthusian growth and Mendelian genetics</li>
<li>Hardy-Weinberg Equilibrium (HWE) at a single locus</li>
<li>Heterozygosity, deleterious alleles, and mutation</li>
<li>Genetics at multiple loci and DNA variation</li>
</ul>
</div>
<div class="section level4">
<h4 id="id_3-genetic-drift-name-popgen_drift">3. Genetic Drift (name: <code>popgen_drift</code>)<a class="anchor" aria-label="anchor" href="#id_3-genetic-drift-name-popgen_drift"></a>
</h4>
<ul>
<li>Intuition and qualitative expectations of genetic drift</li>
<li>Variability in outcomes and heterozygosity decay</li>
<li>Effective population size and historical context</li>
</ul>
</div>
<div class="section level4">
<h4 id="id_4-selection-in-population-genetics-name-popgen_selection">4. Selection in Population Genetics (name: <code>popgen_selection</code>)<a class="anchor" aria-label="anchor" href="#id_4-selection-in-population-genetics-name-popgen_selection"></a>
</h4>
<ul>
<li>Breeding effective population size</li>
<li>Mutation-drift equilibrium and selection</li>
<li>Case studies (e.g., the peppered moths)</li>
</ul>
</div>
<div class="section level4">
<h4 id="id_5-deep-time-molecular-clocks-name-deeptime_clocks">5. Deep-Time Molecular Clocks (name: <code>deeptime_clocks</code>)<a class="anchor" aria-label="anchor" href="#id_5-deep-time-molecular-clocks-name-deeptime_clocks"></a>
</h4>
<ul>
<li>Sequence data and genetic distance</li>
<li>Poisson correction and Jukes-Cantor models</li>
<li>Molecular clocks and their uncertainty</li>
<li>Inferences about deep time</li>
</ul>
</div>
<div class="section level4">
<h4 id="id_6-fossils-and-deep-time-patterns-name-deeptime_rocks">6. Fossils and Deep-Time Patterns (name: <code>deeptime_rocks</code>)<a class="anchor" aria-label="anchor" href="#id_6-fossils-and-deep-time-patterns-name-deeptime_rocks"></a>
</h4>
<ul>
<li>Exploring fossil occurrences and diversity patterns</li>
<li>Spatial distribution and conclusions from fossil records</li>
<li>Dating fossils in absolute time (technical notes)</li>
</ul>
</div>
<div class="section level4">
<h4 id="id_7-birth-death-models-in-deep-time-name-birthdeath_deeptime">7. Birth-Death Models in Deep Time (name: <code>birthdeath_deeptime</code>)<a class="anchor" aria-label="anchor" href="#id_7-birth-death-models-in-deep-time-name-birthdeath_deeptime"></a>
</h4>
<ul>
<li>Deterministic expectations and stochastic processes</li>
<li>Effects of variation on diversification rates</li>
<li>Age-richness models and extinction dynamics</li>
</ul>
</div>
<div class="section level4">
<h4 id="id_8-phylogenies-and-birth-death-models-name-birthdeath_phylogenies">8. Phylogenies and Birth-Death Models (name: <code>birthdeath_phylogenies</code>)<a class="anchor" aria-label="anchor" href="#id_8-phylogenies-and-birth-death-models-name-birthdeath_phylogenies"></a>
</h4>
<ul>
<li>Estimating diversification rates under pure birth and birth-death models</li>
<li>Factors influencing speciation and extinction</li>
</ul>
</div>
</div>
</div>
<div class="section level1">
<h1 id="if-you-never-used-r-before">If you never used R before…<a class="anchor" aria-label="anchor" href="#if-you-never-used-r-before"></a>
</h1>
<p>If you’re new to R, this “pre-vignette” tutorial serves as an introduction to using <code>evolved</code>. It was largely inspired by the first tutorials in EEB 485 and EEB 391/516 (Population and Community Ecology &amp; Evolutionary processes and Macroevolution), but it is extended and modified to work in our context. This document is meant to serve as a first contact with the tools that will be of major importance in our vignettes with <code>evovled</code>. Nothing related to this tutorial is required to use the R package, but we will be using all the tools explained here, so please make sure you have completed it before then (or to be genrlly familiar with R).</p>
<p>This tutorial will be somewhat long, and if this is your first time using R, it will also be a bit confusing. This is because a reasonable part of what you will see here is pretty abstract. We ask you to trust the learning process and keep reading until the end. We highly encourage you to reach out to other peers more knowledgeable in R, if needed. We know that different folks come with very different levels of comfort with R – which means you will learn a lot with your colleagues, however comfortable you may currently be with R.</p>
<div class="section level2">
<h2 id="what-is-r-and-how-can-i-install-it">What is R and how can I install it?<a class="anchor" aria-label="anchor" href="#what-is-r-and-how-can-i-install-it"></a>
</h2>
<p>R is (1) an open source software, (2) a programming language, and (3) a programming environment. The first one means it is a computer program you can freely download, use, or modify to your own needs. The second means it is a language that allows humans and machines to communicate. The third means it gives you everything you need to create your own programs under a specific language. Scientists, especially in the biological sciences, widely use R to run simulations and analyze their datasets, so we will do this too.</p>
<p>If you’re using your own computer and have not yet installed R, you can download and install it for free by pasting this in your web browser:</p>
<p><a href="http://cran.us.r-project.org/" class="external-link uri">http://cran.us.r-project.org/</a></p>
<p>After you download it, install it as you would do with any other program – the installer program will direct you though the major steps. When you open R, you should see something close to Figure 1.</p>
<p>Please download and install R before progressing to install RStudio.</p>
<div class="figure" style="text-align: center">
<img src="README_images/R_screenshot.png" alt="Example of recently-opened R on an Apple computer. The interface looks slightly different on a Windows computer, but the main features are the same. Note that depending on the details, you may or may not see the same warnings and errors as the screenshot." width="100%"><p class="caption">
Example of recently-opened R on an Apple computer. The interface looks slightly different on a Windows computer, but the main features are the same. Note that depending on the details, you may or may not see the same warnings and errors as the screenshot.
</p>
</div>
</div>
<div class="section level2">
<h2 id="what-is-rstudio-and-how-can-i-install-it">What is RStudio and how can I install it?<a class="anchor" aria-label="anchor" href="#what-is-rstudio-and-how-can-i-install-it"></a>
</h2>
<p>RStudio is a development environment for R. As a very coarse simplification, you can think of RStudio as a complement to R that makes some tasks easier – a “shell” enveloping R that makes some communications between you and R easier. However, many would say that RStudio is not much of a helpful resource at all. If you never worked with R before, you are encouraged to try both RStudio and just “plain” R separately and make your own conclusion as to which makes you more comfortable. You are not required to use RStudio to use <code>evolved</code>, but you can use it if you think it will help your learning process (you are required to use R, though).</p>
<p>Here in this class we will use programming as a learning tool, but programming is also a skill that will develop your quantitative ability and will make you a better scientist. If you really want to train this skill and get really good at it, we highly recommend that at some point of your career, you get used to programming just with R – and you are invited to use this class as one opportunity to do that. However, if your intention is to use R only occasionally, it might be worth jumping straight to RStudio, at the cost of being limited by a series of details and constraints that RStudio has.</p>
<p>If you need RStudio for your personal machine, you can download and install it for free by pasting this in your web browser:</p>
<p><a href="http://rstudio.org" class="external-link uri">http://rstudio.org</a></p>
<p>After you download it, install as you would do with any other program – the installer program will direct you though the major steps.</p>
<p>If you intend to use RStudio, please download and install it before progressing to install RMarkdown. We will cover RMarkdown later in this tutorial.</p>
</div>
<div class="section level2">
<h2 id="starting-r-and-rstudio">Starting R and RStudio:<a class="anchor" aria-label="anchor" href="#starting-r-and-rstudio"></a>
</h2>
<p>When you start R, you should see something like figure 1. If you start RStudio instead, it will open R inside of the RStudio environment, and should look like figure 2.</p>
<div class="figure" style="text-align: center">
<img src="README_images/RStudio_screenshot.png" alt="Example of recently-opened Rstudio. Note that the appearance of RStudio is customizable, so your interface might not look the same as the RStudio your instructors or your fellow students have" width="100%"><p class="caption">
Example of recently-opened Rstudio. Note that the appearance of RStudio is customizable, so your interface might not look the same as the RStudio your instructors or your fellow students have
</p>
</div>
<p>Just looking at figures 1 and 2, it is easy to see that RStudio gives a lot more information. We will briefly walk through the items shown in figure 2 to familiarize you with each quadrant in the RStudio environment. Note that in both figures we are not showing the upper part of the screen (containing the tabs “File”, “Edit”, “Code”, “View”, etc.), but sometimes you will use those options too.</p>
<ul>
<li><p>Upper-left: The script. This is where you view, edit, and save text files, including R scripts (.R files) and RMarkdown (.Rmd) files. You will use this quadrant the most during your learning activities.</p></li>
<li><p>Lower-left: the console (a.k.a terminal, a.k.a. command line). This is where lines of script get processed and computed. Here, you can see what code was actually input and run, and what the computer’s output was. You can also paste and execute commands directly into the console, instead of through the script file. This space will also be used exhaustively.</p></li>
<li>
<p>Upper-right: Four tabs:</p>
<ul>
<li>Environment: for loading/saving/examining data (you may or may not use this one much, will depend on your habits)</li>
<li>History: a list of commands you’ve typed (again, use here will depend on your habits)</li>
<li>Connections: for linking to specific database types (we will basically not use this during vignettes)</li>
<li>Tutorial: a useful resource with tutorial demonstrations of R code and functions (not used at all in <code>evolved</code>)</li>
</ul>
</li>
<li>
<p>Lower-right: Five tabs:</p>
<ul>
<li>Files: a list of files in the “working directory” (more on this later in this tutorial)</li>
<li>Plots: where the plots/figures you create will appear (we will use this extensively in vignettes)</li>
<li>Packages: an area for managing installed R packages (probably you will not use it much during vignettes)</li>
<li>Help: access to official documentation for R (we will use this extensively)</li>
<li>Viewer: for viewing web content (probably not used frequently in vignettes)</li>
</ul>
</li>
</ul>
<p>If you want to use RStudio, there are a series of configurations that we highly recommend you have. You can access them by going to the upper part of the screen (the one that contains the tabs “File”, “Edit”, “Code”, “View”, etc), clicking <em>Tools &gt; Global Options</em>, and then editing options. Opinions on the best configuration vary, but below you have the link to a blog post that shows a possible configuration that might be helpful during our classes: (link: <a href="https://www.r-bloggers.com/2019/05/my-rstudio-configuration/" class="external-link uri">https://www.r-bloggers.com/2019/05/my-rstudio-configuration/</a>)</p>
<p>Use your time to get used to all this new software.</p>
</div>
<div class="section level2">
<h2 id="what-is-rmarkdown-and-how-can-i-install-it-optional">What is RMarkdown and how can I install it (optional)?<a class="anchor" aria-label="anchor" href="#what-is-rmarkdown-and-how-can-i-install-it-optional"></a>
</h2>
<p>RMarkdown is a programming language and environment that generates reporting documents. It makes it easy to mix a decently formatted document (and nice-looking equations) with R code, and this is why you want to use RMarkdown when studying ecology and evolutionary biology.</p>
<p>If you know LaTeX, you should be familiar with RMarkdown’s syntax (i.e., the way functions are written and used in a programming language) as they resemble each other. If you don’t know what LaTeX means, no problem – you will have time to familiarize yourself with this kind of tool.</p>
<p>Downloading RStudio automatically downloads RMarkdown too. Otherwise, you can download RMarkdown directly from the R console, by typing.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">'rmarkdown'</span><span class="op">)</span></span>
<span><span class="co"># Using the code above, you will install RMarkdown from The Comprehensive R </span></span>
<span><span class="co"># Archive Network (a.k.a. CRAN).</span></span></code></pre></div>
<p>Please check that you have the <code>knitr</code> package installed in your computer, as RMarkdown needs it.</p>
<p>You can create a new RMarkdown file in RStudio by going to the upper tabs, then clicking on <em>File &gt; New &gt; R Markdown</em>.</p>
<p>You can create .html files from code by clicking on the “Knit” button just above the script window (not shown in figure 2, since the file open there is a script file, .R, not an RMarkdown file, .Rmd).</p>
<div class="section level3">
<h3 id="more-on-rmarkdown">More on RMarkdown<a class="anchor" aria-label="anchor" href="#more-on-rmarkdown"></a>
</h3>
<p>RMarkdown is the result of a philosophy called “literate programming”, first idealized by Donald Knuth during the ’70s. Doctor Knuth believed that programmers should think of programs not only as lists of commands, but also as works of literature. One of his quotes about this approach to coding is <em>“Instead of imagining that our main task is to instruct a computer what to do, let us concentrate rather on explaining to human beings what we want a computer to do”</em>. This has very clear and positive consequences both for teaching and for science, so those who feel comfortable with RMarkdown should follow Dr. Knuth’s ideas when studying using <code>evolved</code>, and use RMarkdown as an intrinsic part of your learning process generally.</p>
<p>Our experience is that it is worthwhile to use RMarkdown to create notes for your scientific practices, such as data analyses, model making/testing, etc. We understand, however, that the learning curve of R (and maybe RStudio) + RMarkdown might be a little to steep. And this is why you can also just give yourself a little time to learn each tool.</p>
<p>This is our recommended routine to work with <code>evolved</code>:</p>
<ol style="list-style-type: decimal">
<li><p>Each lab tutorial is available as vignette in <code>evovled</code>. We recommend skimming the text before starting the tutorial.</p></li>
<li><p>The rest of your studies will be reading the tutorial and answering the questions in it. We highly recommend everyone to read the vignette completely <em>before starting to answer the questions</em>.</p></li>
<li><p>You can use RMarkdown to knit your own answers as PDF files (and this will, with time, prove to be very useful for you in science), but depending on your operational system this might be a bit of a headache. You can also use Microsoft Word or Google Docs, as long as it is usful to you.</p></li>
</ol>
</div>
<div class="section level3">
<h3 id="rmarkdown-advantages">RMarkdown advantages<a class="anchor" aria-label="anchor" href="#rmarkdown-advantages"></a>
</h3>
<p>In addition to plotting and integrating comments with code, RMarkdown also allows you to easily write equations with beautiful notation. Below, we will write down some of the equations we will see in <code>evolved</code>. No need to understand what they mean (biologically speaking) now – I just want to show you the RMarkdown equation-writing feature.</p>
<p>You are likely reading a PDF/html file, but a major point is that this file is built so you can visualize both “sides” of writing a PDF from a RMarkdown file: the “public-facing” side is the PDF file you are currently reading, but the “internal” side is the code that generates the PDF. The .Rmd file that generated this PDF document is posted on <a href="">MJ’s github</a> so that you can see how each of those two files communicate (and how Markdown works “under the hood”).</p>
<p>The point of the following text example is for you to have an idea of how nicely-written equations, code, and text can be merged into a single, organized document through RMarkdown.</p>
<p>Download and open the .Rmd file which has the same name as this PDF (posted on <a href="">MJ’s github</a> to learn the syntax associated with everything we are doing below.</p>
</div>
<div class="section level3">
<h3 id="an-example-of-why-rmarkdown-is-so-neat">An example of why RMarkdown is so neat:<a class="anchor" aria-label="anchor" href="#an-example-of-why-rmarkdown-is-so-neat"></a>
</h3>
<p>The effective population size when there is time-varying population size (i.e., bottlenecks and expansions) is:</p>
<p>The probability of identity-by-descent of two alleles is:</p>
<p>And the rate of heterozygosity loss is:</p>
<p>Which is also equal to <math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi>e</mi><mrow><mi>−</mi><mfrac><mi>t</mi><mrow><mn>2</mn><mi>N</mi></mrow></mfrac></mrow></msup><annotation encoding="application/x-tex">e^{-\frac{t}{2N}}</annotation></semantics></math>. Of course, this assumes a diploid organism, so the census size of this population is <math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>N</mi><annotation encoding="application/x-tex">N</annotation></semantics></math>.</p>
<p>RMarkdown can also very easily merge R plots and text. For instance if we want to graph equation 3, we could simply write:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#create a numeric vector representing time:</span></span>
<span><span class="va">t</span><span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span>from<span class="op">=</span><span class="fl">0</span>, to<span class="op">=</span><span class="fl">200</span>, by<span class="op">=</span><span class="fl">0.001</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#insert population size:</span></span>
<span><span class="va">N</span><span class="op">=</span><span class="fl">20</span></span>
<span></span>
<span><span class="va">Ht_Ho</span><span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">exp</a></span><span class="op">(</span><span class="op">-</span><span class="fl">1</span><span class="op">/</span><span class="op">(</span><span class="fl">2</span><span class="op">*</span><span class="va">N</span><span class="op">)</span><span class="op">)</span><span class="op">^</span><span class="va">t</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">t</span>, y<span class="op">=</span><span class="va">Ht_Ho</span>, type<span class="op">=</span><span class="st">"l"</span>, lwd<span class="op">=</span><span class="fl">4</span>, </span>
<span>     ylab<span class="op">=</span><span class="st">"Rate of heterozygosity loss"</span>,</span>
<span>     xlab<span class="op">=</span><span class="st">"Time"</span>, main<span class="op">=</span><span class="st">"Plot of equation (3)"</span><span class="op">)</span></span></code></pre></div>
<p><img src="README_files/figure-gfm/unnamed-chunk-4-1.png"><!-- --></p>
</div>
<div class="section level3">
<h3 id="rmarkdown-practice-part-i">RMarkdown practice (part I)<a class="anchor" aria-label="anchor" href="#rmarkdown-practice-part-i"></a>
</h3>
<p>Please take a look of the RMarkdown used to make this file (in <a href="">MJ’s github</a> and try to locate the code responsible for each part of what you have just read, putting special attention to codes, plots, and equations.</p>
</div>
<div class="section level3">
<h3 id="rmarkdown-practice-part-ii">RMarkdown practice (part II)<a class="anchor" aria-label="anchor" href="#rmarkdown-practice-part-ii"></a>
</h3>
<p>Now let’s do something which is more hands-on. Create a brand new .Rmd file and then insert text in it by pasting some of the contents of this tutorial to the new file. To run a “test” of one piece of your pasted code, click on a line of your R code in the document and then choose Code &gt; Run Selected Line(s) to run that line in the R console. Alternatively, you can also click the Run button in the top left pane, but the quickest way to do this is the Command+Return on a Mac keyboard, or Control+Return on a Windows keyboard. Test what works best on Linux, if you use it.</p>
<p>Select Run Region &gt; Run Current Chunk to run an entire “chunk” of R code. Save your changes in your .Rmd, and knit to HTML. In the future, note that RStudio automatically saves your .Rmd file whenever you knit it to an HTML or PDF file - but recall that for the purposes of this class we will only accept PDF submissions.</p>
</div>
</div>
<div class="section level2">
<h2 id="working-routine-in-r">Working Routine in R<a class="anchor" aria-label="anchor" href="#working-routine-in-r"></a>
</h2>
<p>In this short tutorial we will help you to navigate through your files. You can think of this as a checklist to use at the start of any of the labs in this class. Steps 1-4 have to be done before any lab. Steps 5 and 6 will be needed depending on the lab.</p>
<p>We start by revising what the “Desktop” of your computer is – the standard folder most computer operating systems (i.e., Windows, MacOS, Linux…) have. It is the first “place” your computer puts you in when you turn it on, seen in the screen shot below, which is my Desktop:</p>
<p><img src="README_images/set_dir_fig00.png" width="100%"></p>
<p>If you are not comfortable with coding or R, we suggest you create a folder for storing all our classwork in the “Desktop” of your computer, like we did in the screen shot above. If you are comfortable with programming, work as you wish.</p>
<div class="section level3">
<h3 id="step-1-setting-work-directory">Step 1: Setting work directory<a class="anchor" aria-label="anchor" href="#step-1-setting-work-directory"></a>
</h3>
<p>The working directory is the folder in your computer that R will navigate to during a session. Any time you import files or write files to a folder, R will automatically use the folder that is set as the working directory. To set the working directory, open R and click, in the upper bar, on <em>Session &gt; Set Working Directory &gt; Choose Directory</em> as in the figure below (notice that my .Rmd file is on the left panel and my R console, different from the RStudio default, is on the right – notice also that I minimized all other panels in my RStudio):</p>
<p><img src="README_images/set_dir_fig0.png" width="100%"></p>
<p>After clicking on <em>Choose Directory</em>, you should see something like the screenshot below:</p>
<p><img src="README_images/set_dir_fig1.png" width="100%"></p>
<p>Once there, you can navigate to the folder where you are storing files for this class, or a subfolder of that folder, to set it as your working directory. It’s important to keep your working folders for each lab organized, as they will include several different files types, including both R scripts (<code>.R</code> files) and data files (e.g., <code>.txt</code>, <code>.csv</code>, etc.). For instance if you had downloaded the files <code>cyt-oxidase1-mt-aligned.txt</code> and <code>R_functions.R</code> and added them to your working directory folder, it would look like this:</p>
<p><img src="README_images/set_dir_fig2.png" width="100%"></p>
<p>After following the above steps, you will have set the directory in which you will be working for the current R “session”. This is the folder R will use as a reference, from now until you close R or change the directory.</p>
</div>
<div class="section level3">
<h3 id="step-2-checking-working-directory">Step 2: Checking working directory<a class="anchor" aria-label="anchor" href="#step-2-checking-working-directory"></a>
</h3>
<p><strong>(I)</strong> To check what working directory you are in, type <code><a href="https://rdrr.io/r/base/getwd.html" class="external-link">getwd()</a></code> into the console or into a script file. This function will tell you which folder is currently set as the working directory by giving you a path name to the folder. If you decided to store your folder in the Desktop, you should see something in the following format (note this follows Mac/Linux formatting, Windows will write this slightly differently):</p>
<p><code>".../[username]/Desktop/[name_of_your_folder]"</code></p>
<p>The things we put inside brackets in the line above are things we expect to be different in your case, given your operational system and the names of your folders.</p>
<p>In my computer (in a Mac/Linux system), I get:</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a><span class="fu">getwd</span>()</span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a>[<span class="dv">1</span>] <span class="st">"/Users/matheus/Desktop/EEB516_labs/lab_05"</span></span></code></pre></div>
<p>R can also return a “compacted” version of the exact same line, which would be:</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a><span class="fu">getwd</span>()</span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a>[<span class="dv">1</span>] <span class="st">"~/Desktop/EEB516_labs/lab_05"</span></span></code></pre></div>
<p>This contraction in particular occurs in Linux and MacOS systems. Windows may show a similar contraction, but it can vary from version to version, and you are welcome to reach out to your GSI in case there are any problems. This contraction is a resource that helps more advanced programmers (at the cost of being slightly confusing for newcomers).</p>
<p>What I recommend you to do is: once you have finished step 1 and your working directory is set, run <code><a href="https://rdrr.io/r/base/getwd.html" class="external-link">getwd()</a></code> and paste its output into the <code><a href="https://rdrr.io/r/base/getwd.html" class="external-link">setwd()</a></code> function. Insert that line in your “setup” code chunk. This will help RMarkdown to compile more smoothly. It will look more or less like the following screenshot:</p>
<p><img src="README_images/set_dir_fig3.png" width="100%"></p>
<p>Another strange behavior that RMarkdown might do is to automatically change your folder after every chunk of code you run. This is extremely counterproductive, and in the end (step 7) we provide a guide on a possible way to solve this.</p>
<p><strong>(II)</strong> To check what files are inside your working directory, type <code><a href="https://rdrr.io/r/base/list.files.html" class="external-link">dir()</a></code> into the console or into a script file. This will list the names of every file or subfolder within the working directory. For example, if I had two files in my working directory named <code>cyt-oxidase1-mt-aligned.txt</code> and <code>R_functions.R</code>, I would see the following output after typing <code><a href="https://rdrr.io/r/base/list.files.html" class="external-link">dir()</a></code>:</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a>[<span class="dv">1</span>] <span class="st">"cyt-oxidase1-mt-aligned.txt"</span></span>
<span id="cb8-2"><a href="#cb8-2" tabindex="-1"></a>[<span class="dv">2</span>] <span class="st">"R_functions.R"</span></span></code></pre></div>
<p><strong>(III)</strong> To check what R objects are currently in your workspace, type <code><a href="https://rdrr.io/r/base/ls.html" class="external-link">ls()</a></code>, which gives you a list of all objects currently loaded in your R environment. Before you start your lab, the results should always be <code>character(0)</code>, which means there is currently nothing in your workspace – we call this a “cleaned” workspace.</p>
<p>The above steps should have helped you set everything and check what files and objects are available to you. Below are some troubleshooting tips in case things went awry in the previous steps.</p>
<p><strong>IF</strong> your result from <code><a href="https://rdrr.io/r/base/getwd.html" class="external-link">getwd()</a></code> is different from what it should be, try step 1 (setting your work directory) again.</p>
<p><strong>IF</strong> your result from <code><a href="https://rdrr.io/r/base/list.files.html" class="external-link">dir()</a></code> is different from what it should be, you could either be in the wrong folder (so go back to step 1) or you could not have all the files in the correct folder – correct this by moving files in your computer.</p>
<p><strong>IF</strong> ls() returns any other thing than <code>"character(0)"</code> you should clean your workspace in R by typing <code>rm(list=ls())</code> in the console, but be careful as this will erase everything you have previously created within R (although things can be preserved if you previously have saved them in a file external to R such as a .txt or .csv file).</p>
<p>Having a clean workspace is very important to working properly in R – many problems arise from forgetting to start from a completely clean workspace.</p>
</div>
<div class="section level3">
<h3 id="step-3-saving-files">Step 3: Saving files<a class="anchor" aria-label="anchor" href="#step-3-saving-files"></a>
</h3>
<p>Make sure <em>all files</em> are being saved correctly. This is important to not lose your work if R crashes, class time ends, or you have to stop for any other reason. Go to the upper bar and click on <em>File &gt; Save As</em> as in the figure below.</p>
<p><img src="README_images/set_dir_fig5.png" width="100%"></p>
<p>A window will pop up in the folder you are working in. You can save your .Rmd file there, as you would do with any other file.</p>
<p>Now, if you run <code><a href="https://rdrr.io/r/base/list.files.html" class="external-link">dir()</a></code> again, you should see your saved file. As I saved my file with the name <code>lab_05_answers.Rmd</code>, I got:</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a>[<span class="dv">1</span>] <span class="st">"cyt-oxidase1-mt-aligned.txt"</span></span>
<span id="cb9-2"><a href="#cb9-2" tabindex="-1"></a>[<span class="dv">2</span>] <span class="st">"lab_05_answers.Rmd"</span>         </span>
<span id="cb9-3"><a href="#cb9-3" tabindex="-1"></a>[<span class="dv">3</span>] <span class="st">"R_functions.R"</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="step-4-compilingknitting-your-rmarkdown">Step 4: Compiling/knitting your RMarkdown<a class="anchor" aria-label="anchor" href="#step-4-compilingknitting-your-rmarkdown"></a>
</h3>
<p>Now is the time to make sure that your RMarkdown compiles. You can compile it by clicking just above the script window in RStudio on <code>Knit</code>:</p>
<p><img src="README_images/set_dir_fig6.png" width="100%"></p>
<p>Now, if you run <code><a href="https://rdrr.io/r/base/list.files.html" class="external-link">dir()</a></code> again, you should see your new compiled <code>.html</code> document in the working folder:</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a>[<span class="dv">1</span>] <span class="st">"cyt-oxidase1-mt-aligned.txt"</span></span>
<span id="cb10-2"><a href="#cb10-2" tabindex="-1"></a>[<span class="dv">2</span>] <span class="st">"lab_05_answers.html"</span>         </span>
<span id="cb10-3"><a href="#cb10-3" tabindex="-1"></a>[<span class="dv">3</span>] <span class="st">"lab_05_answers.Rmd"</span>         </span>
<span id="cb10-4"><a href="#cb10-4" tabindex="-1"></a>[<span class="dv">4</span>] <span class="st">"R_functions.R"</span></span></code></pre></div>
<p><strong>Note:</strong> I strongly recommend you <em>always</em> knit/compile your <code>.Rmd</code> after finishing each question. There is nothing more disappointing than finishing an RMarkdown and then not being able to compile it. So knitting along the way will eliminate this possibility.</p>
</div>
<div class="section level3">
<h3 id="step-5-sourcing-r-scripts">Step 5: Sourcing R scripts<a class="anchor" aria-label="anchor" href="#step-5-sourcing-r-scripts"></a>
</h3>
<p>You might need to source files – which is just reading an .R file, importing any functions or data from that script file to your current script file. All your files should be in the folder where you are working, so if you followed previous steps, you just have to type: <code>source("name_of_file.R")</code> (“name_of_file” is merely a placeholder).</p>
<p>If there are no errors after typing this in, the script you just sourced ran until the end with no issue. If your sourced file crashed or printed an error, something is wrong, so please revise all steps above and if things continue to not work, ask your GSI.</p>
</div>
<div class="section level3">
<h3 id="step-6-loading-external-data">Step 6: Loading external data<a class="anchor" aria-label="anchor" href="#step-6-loading-external-data"></a>
</h3>
<p>You might need to load external files, such as <code>.csv</code> files. We will probably tell you, in each tutorial, which function to use. But the general format would be:</p>
<p><code>object &lt;- function_that_reads_file("file_name")</code></p>
<p>For instance, if we had a spreadsheet called <code>data.csv</code> to read in and we have correctly followed all the steps above, we would run:</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" tabindex="-1"></a>data <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"data.csv"</span>)</span></code></pre></div>
<p>Too check that the data was read correctly, we could do something like this:</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" tabindex="-1"></a><span class="fu">head</span>(data)</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="step-7-debbuging-one-specific-strange-behavior-some-students-found">Step 7: Debbuging one specific strange behavior some students found<a class="anchor" aria-label="anchor" href="#step-7-debbuging-one-specific-strange-behavior-some-students-found"></a>
</h3>
<p>A strange behavior RMarkdown might have is to automatically change your working directory folder after every chunk of code you run. This is extremely counterproductive, but you can fix it by going to the upper bar and clicking <em>Tools &gt; Global Options</em> as in the figure below:</p>
<p><img src="README_images/set_dir_fig4.png" width="100%"></p>
<p>Then go to <em>RMarkdown &gt; Evaluate chunks in directory</em> and set the option to “Document”, as in the figure below:</p>
<p><img src="README_images/set_dir_fig4.5.png" width="100%"></p>
<p>The configurations you see in the screenshot above are used by Matheus – feel free to try them as well if you want.</p>
<p>If RStudio keeps changing the directory <em>after you have done the steps above</em>, reach out to your GSI. Other ways to try to solve any problem you may encounter is to look it up on Google, as R and RStudio have a very big, helpful, and kind online community that usually helps with all kinds of problems – all of us regularly troubleshoot things by looking them up online.</p>
</div>
</div>
</div>

  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://cloud.r-project.org/package=evolved" class="external-link">View on CRAN</a></li>
<li><a href="https://github.com/mjanuario/evolved/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/mjanuario/evolved/issues" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small>GPL (&gt;= 3)</small></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing evolved</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Matheus Januario <br><small class="roles"> Author, copyright holder, maintainer </small> <a href="https://orcid.org/0000-0002-6480-7095" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Jennifer Auler <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0001-7576-9058" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Andressa Viol <br><small class="roles"> Author </small>  </li>
<li>Daniel Rabosky <br><small class="roles"> Author </small>  </li>
</ul>
</div>



  </div>
</div>


      <footer><div class="copyright">
  <p></p>
<p>Developed by Matheus Januario, Jennifer Auler, Andressa Viol, Daniel Rabosky.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

      </footer>
</div>






  </body>
</html>
