<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Tools for population genetics • evolved</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Tools for population genetics">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">evolved</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/birthdeath_deeptime.html">Birth-death models and deep time</a>
    </li>
    <li>
      <a href="../articles/birthdeath_phylogenies.html">Birth-death models and molecular phylogenies</a>
    </li>
    <li>
      <a href="../articles/deeptime_clocks.html">Allele frequency change and selection</a>
    </li>
    <li>
      <a href="../articles/deeptime_rocks.html">Introduction to the fossil record</a>
    </li>
    <li>
      <a href="../articles/intro_R.html">Introduction to R</a>
    </li>
    <li>
      <a href="../articles/phylogenies_and_diversification.html">Phylogenies and diversification</a>
    </li>
    <li>
      <a href="../articles/popgen_drift.html">Sampling alleles, genetic drift and effective population size</a>
    </li>
    <li>
      <a href="../articles/popgen_intro.html">Introduction to evolutionary genetics</a>
    </li>
    <li>
      <a href="../articles/popgen_selection.html">Allele frequency change and selection</a>
    </li>
    <li>
      <a href="../articles/population_genetics.html">Tools for population genetics</a>
    </li>
    <li>
      <a href="../articles/rocks_and_clocks.html">Clocks and Rocks</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/mjanuario/evolved/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Tools for population genetics</h1>
                        <h4 data-toc-skip class="author">Matheus
Januario and Jennifer Auler</h4>
            
            <h4 data-toc-skip class="date">Oct/2023</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/mjanuario/evolved/blob/HEAD/vignettes/population_genetics.Rmd" class="external-link"><code>vignettes/population_genetics.Rmd</code></a></small>
      <div class="hidden name"><code>population_genetics.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">evolved</span><span class="op">)</span></span></code></pre></div>
<div class="section level2">
<h2 id="hardy-weinberg-equilibrium">Hardy-Weinberg equilibrium<a class="anchor" aria-label="anchor" href="#hardy-weinberg-equilibrium"></a>
</h2>
<p>The Hardy-Weinberg equilibrium (HWE) is a deterministic model for the
maintenance of allele-level biodiversity. However, stochastic thinking
is already well-established as an important aspect of studying
evolutionary biology - and so there are clear advantages in introducing
this kind of thinking to students.</p>
<p>However, typical HWE equations or study material related to it do not
address stochastic variation coming from finite numbers of individuals.
The function <code><a href="../reference/OneGenHWSim.html">OneGenHWSim()</a></code> addresses this issue. With it,
students can graph and explore stochasticity under HWE:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">nsims</span> <span class="op">=</span> <span class="fl">1000</span></span>
<span><span class="va">sims</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/OneGenHWSim.html">OneGenHWSim</a></span><span class="op">(</span>n.ind <span class="op">=</span> <span class="fl">57</span>, p <span class="op">=</span> <span class="fl">.2</span>, n.sim <span class="op">=</span> <span class="va">nsims</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/boxplot.html" class="external-link">boxplot</a></span><span class="op">(</span><span class="va">sims</span>, frame.plot<span class="op">=</span><span class="cn">F</span>,</span>
<span>        ylab<span class="op">=</span><span class="st">"Number of individuals"</span>, xlab<span class="op">=</span><span class="st">"Genotype"</span><span class="op">)</span></span>
<span><span class="kw">for</span><span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">sims</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/graphics/lines.html" class="external-link">lines</a></span><span class="op">(</span>x<span class="op">=</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span>, y<span class="op">=</span><span class="va">sims</span><span class="op">[</span><span class="va">i</span>, <span class="op">]</span>, col<span class="op">=</span><span class="st">"#A020F064"</span><span class="op">)</span>  </span>
<span><span class="op">}</span></span></code></pre></div>
<p><img src="population_genetics_files/figure-html/unnamed-chunk-3-1.png" width="576" style="display: block; margin: auto;"></p>
<p>Of course, the object generated by the simulations can also be
manipulated as if it was “data”</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">sims</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##       A1A1 A1A2 A2A2</span></span>
<span><span class="co">## sim_1    2   18   37</span></span>
<span><span class="co">## sim_2    5   12   40</span></span>
<span><span class="co">## sim_3    1   20   36</span></span>
<span><span class="co">## sim_4    2   18   37</span></span>
<span><span class="co">## sim_5    4   14   39</span></span>
<span><span class="co">## sim_6    3   16   38</span></span></code></pre>
<p>With this type of object, students can also simulate closed
populations of a specific size, thus for instance testing (using
e.g. chi-square tests) actual hypotheses about HWE adequacy to explain a
certain genotypic frequency.</p>
</div>
<div class="section level2">
<h2 id="genetic-drift">Genetic drift<a class="anchor" aria-label="anchor" href="#genetic-drift"></a>
</h2>
<p>Students can also simulate genetic drift in Wright-Fisher
populations, by using the function <code><a href="../reference/WFDriftSim.html">WFDriftSim()</a></code></p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">drifting</span> <span class="op">=</span> <span class="fu"><a href="../reference/WFDriftSim.html">WFDriftSim</a></span><span class="op">(</span>Ne <span class="op">=</span> <span class="fl">10</span>, n.gen <span class="op">=</span> <span class="fl">100</span>, p0 <span class="op">=</span> <span class="fl">.5</span>, n.sim <span class="op">=</span> <span class="fl">1000</span>, </span>
<span>                        print.data <span class="op">=</span> <span class="cn">T</span>, plot.type <span class="op">=</span> <span class="st">"static"</span>, knitr <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="population_genetics_files/figure-html/unnamed-chunk-5-1.png" width="576" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#View the first 10 generation of a few simulations.</span></span>
<span><span class="va">drifting</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">6</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">11</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">##      gen0 gen1 gen2 gen3 gen4 gen5 gen6 gen7 gen8 gen9 gen10</span></span>
<span><span class="co">## sim1  0.5 0.60 0.65 0.75  0.6 0.70 0.70 0.80 0.90 0.90  0.80</span></span>
<span><span class="co">## sim2  0.5 0.25 0.25 0.35  0.6 0.70 0.80 0.80 0.85 0.75  0.85</span></span>
<span><span class="co">## sim3  0.5 0.40 0.30 0.35  0.4 0.35 0.35 0.35 0.25 0.30  0.40</span></span>
<span><span class="co">## sim4  0.5 0.65 0.65 0.50  0.5 0.55 0.40 0.50 0.45 0.30  0.25</span></span>
<span><span class="co">## sim5  0.5 0.40 0.50 0.50  0.4 0.55 0.40 0.45 0.55 0.30  0.30</span></span>
<span><span class="co">## sim6  0.5 0.40 0.35 0.20  0.2 0.10 0.10 0.05 0.00 0.00  0.00</span></span></code></pre>
<p>The argument <code>printData = T</code> creates an object that can
also be handled as it was “data”, so students can use it to calculate
drift processes, for instance, the amount of time needed until
fixation:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fixchecker</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">res</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/warning.html" class="external-link">suppressWarnings</a></span><span class="op">(</span></span>
<span>      <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">min</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">x</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fl">0</span><span class="op">:</span><span class="fl">1</span><span class="op">)</span>, na.rm <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span>
<span>    <span class="op">)</span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span>  </span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/hist.html" class="external-link">hist</a></span><span class="op">(</span>x <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span>X <span class="op">=</span> <span class="va">drifting</span>, MARGIN <span class="op">=</span> <span class="fl">1</span>, <span class="va">fixchecker</span><span class="op">)</span>,</span>
<span>     main<span class="op">=</span><span class="st">"Time until fixation"</span>, xlab <span class="op">=</span> <span class="st">"Fixation time"</span><span class="op">)</span></span></code></pre></div>
<p><img src="population_genetics_files/figure-html/unnamed-chunk-6-1.png" width="576" style="display: block; margin: auto;"></p>
<p>Or other types of manipulations or measurements.</p>
</div>
<div class="section level2">
<h2 id="natural-selection">Natural selection<a class="anchor" aria-label="anchor" href="#natural-selection"></a>
</h2>
<p>Just like the previously mentioned functions,
<code><a href="../reference/NatSelSim.html">NatSelSim()</a></code> can generate objects that can be handled by
students as if they were “data”.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">selected</span> <span class="op">=</span> <span class="fu"><a href="../reference/NatSelSim.html">NatSelSim</a></span><span class="op">(</span>w11 <span class="op">=</span> <span class="fl">.2</span>, w12 <span class="op">=</span> <span class="fl">.3</span>, w22 <span class="op">=</span> <span class="fl">.1</span>,</span>
<span>            n.gen <span class="op">=</span> <span class="fl">20</span>, print.data <span class="op">=</span> <span class="cn">T</span>, plot.type <span class="op">=</span> <span class="st">"static"</span>, knitr<span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="population_genetics_files/figure-html/unnamed-chunk-7-1.png" width="576" style="display: block; margin: auto;"></p>
<p>Note that by default the function provides 4 animated plots, but this
can be changed using the argument <code>plot_type</code> (see function
help for details).</p>
<p>Now, we can ask: Assuming we simulated the dynamics for enough time,
which genotype in the simulation above reached its equilibrium frequency
faster?</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># a numeric tolerance to accept some frequency as close enough to equilibrium</span></span>
<span><span class="va">tol</span> <span class="op">=</span> <span class="fl">0.0001</span> </span>
<span></span>
<span><span class="va">aux</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/MathFun.html" class="external-link">abs</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="va">selected</span>, <span class="fl">2</span>, <span class="va">diff</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># answer (in number of generations)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="va">aux</span><span class="op">&gt;</span><span class="va">tol</span>, <span class="fl">2</span>, <span class="va">which.min</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## AA Aa aa </span></span>
<span><span class="co">## 20 18 18</span></span></code></pre>
<p>In the above result, we are tabulating the number of dominant
homozygotes (“AA”), hetrozygotes (“Aa”), and recessive homozygotes
(“aa”). Students can manipulate the objects freely, and thus educators
can allow them to use simulations to explore relevant scenarios in an
investigative manner (see types of inquiries in Banchi &amp; Bell,
2008)</p>
</div>
<div class="section level2">
<h2 id="references">References:<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<p>Banchi, H., &amp; Bell, R. (2008). The many levels of inquiry.
Science and children, 46(2), 26.</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Matheus Januario, Jennifer Auler, Andressa Viol, Daniel Rabosky.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

      </footer>
</div>






  </body>
</html>
