<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>DNA sequence evolution • evolved</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="DNA sequence evolution">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">evolved</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">1.0.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../index.html"><span class="fa fa-home"></span> Home</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html"><span class="fa fa-book"></span> Reference</a></li>
<li class="active nav-item"><a class="nav-link" href="../articles/index.html"><span class="fa fa-book-open"></span> Vignettes &amp; Tutorials</a></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html"><span class="fa fa-newspaper"></span> News</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/mjanuario/evolved" aria-label="GitHub repository"><span class="fa fa-github"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>DNA sequence evolution</h1>
                        <h4 data-toc-skip class="author">Matheus
Januario, Andressa Viol, and Daniel Rabosky</h4>
            
            <h4 data-toc-skip class="date">Jan 2024</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/mjanuario/evolved/blob/v1.0.12/vignettes/deeptime_clocks.Rmd" class="external-link"><code>vignettes/deeptime_clocks.Rmd</code></a></small>
      <div class="d-none name"><code>deeptime_clocks.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="learning-objectives">Learning objectives<a class="anchor" aria-label="anchor" href="#learning-objectives"></a>
</h2>
<ol style="list-style-type: decimal">
<li>Exploring sequence data</li>
<li>A simple measure of genetic distance</li>
<li>The Poisson correction</li>
<li>Building a molecular clock</li>
<li>Molecular clocks and inferences about deep time</li>
<li>Jukes-Cantor correction</li>
<li>The uncertainty of your molecular clock</li>
</ol>
</div>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#loading packages we will need today:</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/emmanuelparadis/ape" class="external-link">ape</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/mjanuario/evolved" class="external-link">evolved</a></span><span class="op">)</span></span></code></pre></div>
<p>We have encountered many processes that change allele frequency in a
population, but today we will study the actual genetic differences among
species. We will also learn how to use that genetic information to
reconstruct the evolutionary past of lineages with the so-called
“molecular clock”. To do that, we will use <em>real data</em> on
molecules to estimate the divergence (in numbers of substitutions)
between homologous proteins in different organisms.</p>
<p>Homologous sequences, together with information from the fossil
record, will be used to “calibrate” the rate at which amino acid
substitutions occur during protein evolution. More specifically, given
an estimate of the amount of divergence, plus an estimate of the rate at
which divergence occurs, we can estimate how much time has passed since
divergence.</p>
</div>
<div class="section level2">
<h2 id="exploring-sequence-data">Exploring sequence data<a class="anchor" aria-label="anchor" href="#exploring-sequence-data"></a>
</h2>
<p>We have assembled a set of protein sequences from the cytochrome
oxidase 1 gene. This gene, often known as CO1 (“see-oh-one”), is a
mitochondrial gene that plays a key role in cellular respiration (e.g.,
the primary aerobic pathway to energy or ATP generation). CO1 contains
approximately 513 amino acids and has been used in many studies to
construct phylogenetic trees and estimate divergence times between
taxa.</p>
<p>We manually downloaded full-length genes of a number of focal taxa
from the public database <a href="https://www.ncbi.nlm.nih.gov/genbank/" class="external-link">GenBank</a> (more general
info about GenBank can be found in Benson et al., 2012). We then
<em>aligned</em> the sequences using <a href="https://www.ebi.ac.uk/jdispatcher/msa/clustalo" class="external-link">CLUSTAL</a> (more
info in Sievers et al., 2011), a widely-used sequence alignment
tool.</p>
<p>Because this dataset is included in our R package, we can access it
by simply typing:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"cytOxidase"</span><span class="op">)</span></span></code></pre></div>
<p>Now we can pull out the sequence of a single organism by typing:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cytOxidase</span><span class="op">[</span><span class="st">"human"</span><span class="op">]</span></span></code></pre></div>
<p>Now, let’s explore our sequences a little.</p>
<p>How many sequences do we have?</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">cytOxidase</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 17</span></span></code></pre>
<p>What are the organisms in our dataset?</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">cytOxidase</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##  [1] "cnidaria"   "snake"      "echinoderm" "mollusk"    "alligator" </span></span>
<span><span class="co">##  [6] "lamprey"    "shark"      "bird"       "frog"       "fish"      </span></span>
<span><span class="co">## [11] "platypus"   "human"      "chimpanzee" "bryozoa"    "annelid"   </span></span>
<span><span class="co">## [16] "insect"     "crustacea"</span></span></code></pre>
<p>How long is each sequence?</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">cytOxidase</span>, <span class="va">nchar</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##   cnidaria      snake echinoderm    mollusk  alligator    lamprey      shark </span></span>
<span><span class="co">##        513        513        513        513        513        513        513 </span></span>
<span><span class="co">##       bird       frog       fish   platypus      human chimpanzee    bryozoa </span></span>
<span><span class="co">##        513        513        513        513        513        513        513 </span></span>
<span><span class="co">##    annelid     insect  crustacea </span></span>
<span><span class="co">##        513        513        513</span></span></code></pre>
<p>You can access any particular sequence by typing:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cytOxidase</span><span class="op">[</span><span class="st">"platypus"</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">##                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            platypus </span></span>
<span><span class="co">## "NRWLFSTNHKDIGTLYLLFGAWAGMAGTALSILIRSELGQPGSLLGDDQIYNVIVTAHAFVMIFFMVMPIMIGGFGNWLVPLMIGAPDMAFPRMNNMSFWLLPPSFLLLLVSSTVEAGAGTGWTVYPPLAGNLAHAGASVDLAIFSLHLAGVSSILGAINFITTIINMKPPAMSQYQTPLFVWSVLITAVLLLLSLPVLAAGITMLLTDRNLNTTFFDPAGGGDPILYQHLFWFFGHPEVYILILPGFGIISHIVTYYSGKKEPFGYMGMVWAMMSIGFLGFIVWAHHMFTVGMDVDTRAYFTSATMIIAIPTGVKVFSWLATLHGGDIKWTPPMLWALGFIFLFTVGGLTGIVLANSSLDIILHDTYYVVAHFHYVLSMGAVFAIMGGFVHWFPLLSGFTLHPTWAKVHFTLMFVGVNLTFFPQHFLGLAGMPRRYSDYPDAYTLWNALSSLGSFVSLTAVMVMIFMIWEAFASKREVLS--VELTTTNIEWLHGC-PPPYHTFEQPVYIKA"</span></span></code></pre>
<p>You can compute the number of differences by typing:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/countSeqDiffs.html">countSeqDiffs</a></span><span class="op">(</span><span class="va">cytOxidase</span>, <span class="st">"human"</span>, <span class="st">"alligator"</span><span class="op">)</span>  </span></code></pre></div>
<pre><code><span><span class="co">## [1] 77</span></span></code></pre>
<p>Finally, you can get the length of a specific sequence by typing:</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/nchar.html" class="external-link">nchar</a></span><span class="op">(</span><span class="va">cytOxidase</span><span class="op">[</span><span class="st">"platypus"</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## platypus </span></span>
<span><span class="co">##      513</span></span></code></pre>
<hr>
</div>
<div class="section level2">
<h2 id="a-simple-measure-of-genetic-distance">A simple measure of genetic distance<a class="anchor" aria-label="anchor" href="#a-simple-measure-of-genetic-distance"></a>
</h2>
<p>As mentioned in class, the simplest genetic distance between two
sequences is just the proportion of differences (i.e., the
p-distance).</p>
<p>Now, lets focus on these pairs of taxa:</p>
<pre><code><span><span class="va">human</span><span class="op">-</span><span class="va">bird</span></span>
<span><span class="va">human</span><span class="op">-</span><span class="va">chimp</span></span>
<span><span class="va">human</span><span class="op">-</span><span class="va">platypus</span></span>
<span><span class="va">human</span><span class="op">-</span><span class="va">echinoderm</span></span>
<span><span class="va">cnidaria</span><span class="op">-</span><span class="va">insect</span></span>
<span><span class="va">insect</span><span class="op">-</span><span class="va">crustacean</span></span>
<span><span class="va">annelid</span><span class="op">-</span><span class="va">insect</span></span>
<span><span class="va">chimpanzee</span><span class="op">-</span><span class="va">insect</span></span>
<span><span class="va">chimpanzee</span><span class="op">-</span><span class="va">fish</span></span>
<span><span class="va">platypus</span><span class="op">-</span><span class="va">cnidaria</span></span>
<span><span class="va">frog</span><span class="op">-</span><span class="va">lamprey</span></span>
<span><span class="va">snake</span><span class="op">-</span><span class="va">shark</span></span>
<span><span class="va">insect</span><span class="op">-</span><span class="va">mollusk</span></span>
<span><span class="va">bird</span><span class="op">-</span><span class="va">alligator</span></span></code></pre>
<p>If you are unfamiliar with creating matrices or named vectors in R,
below is a tutorial to walk you through how to do this.</p>
<p>To make a named vector, you will access the <code>names</code>
attribute of a vector with the function <code><a href="https://rdrr.io/r/base/names.html" class="external-link">names()</a></code>. You will
assign another vector that contains the names to the <code>names</code>
attribute of your vector like this:</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">seq_diff_vec</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">4</span>, <span class="fl">8</span>, <span class="fl">3</span>, <span class="fl">6</span>, <span class="fl">0</span><span class="op">)</span> <span class="co"># note that these numbers are completely made up</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">seq_diff_vec</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"snake-chimp"</span>, <span class="st">"human-echinoderm"</span>, <span class="st">"alligator-insect"</span>, </span>
<span>                         <span class="st">"chimp-echinoderm"</span>, <span class="st">"human-chimp"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">seq_diff_vec</span></span></code></pre></div>
<pre><code><span><span class="co">##      snake-chimp human-echinoderm alligator-insect chimp-echinoderm </span></span>
<span><span class="co">##                4                8                3                6 </span></span>
<span><span class="co">##      human-chimp </span></span>
<span><span class="co">##                0</span></span></code></pre>
<p>Note that you cannot repeat names, each name must be unique.</p>
<p>As seen above, you now have a vector whose values are labeled with a
unique identifier. As you have done many times in these labs, you can
also access a particular value within your vector by its unique name
using this format:</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">seq_diff_vec</span><span class="op">[</span><span class="st">"snake-chimp"</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## snake-chimp </span></span>
<span><span class="co">##           4</span></span></code></pre>
<p>To make a matrix, use the handy function <code><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix()</a></code>. Say I
wanted to build a matrix that counts differences between 3 pairs:
snake-human, insect-snake, and insect-human. First, I will build a
skeleton matrix to hold this data:</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">m</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span>data <span class="op">=</span> <span class="cn">NA</span>,  <span class="co"># placeholder value for each matrix's cell will be NA </span></span>
<span>                   <span class="co"># (later will be replaced by real data)</span></span>
<span>       nrow <span class="op">=</span> <span class="fl">3</span>,  <span class="co"># one row for each taxon group</span></span>
<span>       ncol <span class="op">=</span> <span class="fl">3</span>,  <span class="co"># one col for each taxon group</span></span>
<span>       dimnames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"snake"</span>, <span class="st">"human"</span>, <span class="st">"insect"</span><span class="op">)</span>, <span class="co"># these are the row names</span></span>
<span>                       <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"snake"</span>, <span class="st">"human"</span>, <span class="st">"insect"</span><span class="op">)</span><span class="op">)</span> <span class="co"># these are the column names</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>Let’s take a look at our skeleton matrix.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">m</span></span></code></pre></div>
<pre><code><span><span class="co">##        snake human insect</span></span>
<span><span class="co">## snake     NA    NA     NA</span></span>
<span><span class="co">## human     NA    NA     NA</span></span>
<span><span class="co">## insect    NA    NA     NA</span></span></code></pre>
<p>Looks good! Now, to fill in the cell values that currently say
<code>NA</code> with data containing the sequence differences between
each pair, I can directly subset the matrix at the row, column, or cell
that I want using base subsetting in R. A very important thing to note
is that this matrix is symmetric, meaning that both the upper and lower
triangles are the same. It is symmetric because there are two cells for
each pair of taxa, but only one value of sequence differences for that
pair. Thus, you may choose to either leave one half of the matrix blank,
or repeat the values.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">m</span><span class="op">[</span><span class="fl">1</span>,<span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">5</span>, <span class="fl">2</span><span class="op">)</span> <span class="co"># replacing the entire first row with fake numbers</span></span>
<span><span class="va">m</span><span class="op">[</span><span class="fl">2</span>,<span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">5</span>, <span class="fl">0</span>, <span class="fl">80</span><span class="op">)</span> <span class="co"># replacing the entire second row with fake numbers</span></span>
<span><span class="va">m</span><span class="op">[</span><span class="fl">3</span>,<span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2</span>, <span class="fl">80</span>, <span class="fl">0</span><span class="op">)</span> <span class="co"># repalcing the entire third row with fake numbers</span></span>
<span><span class="va">m</span></span></code></pre></div>
<pre><code><span><span class="co">##        snake human insect</span></span>
<span><span class="co">## snake      0     5      2</span></span>
<span><span class="co">## human      5     0     80</span></span>
<span><span class="co">## insect     2    80      0</span></span></code></pre>
<p>Like this, we can manually alter the matrix. You can also do this
with a nested loop, though this is more complicated.</p>
</div>
<div class="section level2">
<h2 id="the-poisson-correction">The Poisson correction<a class="anchor" aria-label="anchor" href="#the-poisson-correction"></a>
</h2>
<p>Here are some extra hints for Q8:</p>
<p>Hint (1): Make sure you have entered your data correctly! A good
thing to always do is to make sure the lengths of your vectors are the
same:</p>
<pre><code><span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">p_dist</span><span class="op">)</span> <span class="co"># assuming your p-distance vector is named like this</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">poiss_dist</span><span class="op">)</span> <span class="co"># same as above for the poisson-corrected distance</span></span></code></pre>
<p>Hint (2): You can add a 1:1 line (perfect correspondence between
axes) by typing in the following <em>after</em> the code that creates
your scatterplot:</p>
<pre><code><span><span class="fu"><a href="https://rdrr.io/r/graphics/abline.html" class="external-link">abline</a></span><span class="op">(</span>a<span class="op">=</span><span class="fl">0</span>, b<span class="op">=</span><span class="fl">1</span><span class="op">)</span> <span class="co"># note that the arguments are the parameters of the line equation</span></span></code></pre>
<div class="section level3">
<h3 id="some-food-for-thought">Some food for thought:<a class="anchor" aria-label="anchor" href="#some-food-for-thought"></a>
</h3>
<p>What is your estimate of the average number of substitutions per site
that have occurred since humans and chimpanzees shared a common
ancestor? What assumption did you make to get this estimate?</p>
<hr>
</div>
</div>
<div class="section level2">
<h2 id="building-a-molecular-clock">Building a molecular clock<a class="anchor" aria-label="anchor" href="#building-a-molecular-clock"></a>
</h2>
<p>Now, we are going to calibrate a molecular clock and apply it to the
distances you have computed above. The <em>rate</em> at which our
metaphorical clock “ticks” will be used to compute the expected number
of substitutions per million years. Just for reference, here is a
phylogeny of all the species in your dataset:</p>
<p><img src="deeptime_clocks_files/figure-html/unnamed-chunk-15-1.png" class="r-plt" alt="" width="700"></p>
<p>Note that we manually set all branches to have equal lengths in the
figure above. What we want now is to make the branches proportional to
time.</p>
<p>For our calibration, we are going to use a single fossil to calibrate
the age of the most-recent ancestor of humans and birds. Fortunately,
there is an extremely useful fossil taxon, <em>Hylonomus</em> known from
Nova Scotia, that is thought to be extremely closely related to the
common ancestor of birds and mammals. Researchers have used the age of
this fossil (312 Ma) to calibrate molecular clocks.</p>
<p>Now, we will use your Poisson-corrected distance between birds and
humans and the dating of to estimate the rate of protein evolution (in
substitutions per site per million years).</p>
<div class="section level3">
<h3 id="molecular-clocks-and-inferences-about-deep-time">Molecular clocks and inferences about deep-time<a class="anchor" aria-label="anchor" href="#molecular-clocks-and-inferences-about-deep-time"></a>
</h3>

</div>
</div>
<div class="section level2">
<h2 id="the-jukes-cantor-correction">The Jukes-Cantor correction<a class="anchor" aria-label="anchor" href="#the-jukes-cantor-correction"></a>
</h2>
<div class="section level4">
<h4 id="some-food-for-thought-1">Some food for thought:<a class="anchor" aria-label="anchor" href="#some-food-for-thought-1"></a>
</h4>
<p>What happens with the Jukes-Cantor distance if
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mo>≥</mo><mn>0.75</mn></mrow><annotation encoding="application/x-tex">p \geq 0.75</annotation></semantics></math>?
Why would that happen?</p>
<p>Hint: Think about what “multiple hits” means.</p>
<hr>
</div>
</div>
<div class="section level2">
<h2 id="the-uncertainty-of-your-molecular-clock">The uncertainty of your molecular clock<a class="anchor" aria-label="anchor" href="#the-uncertainty-of-your-molecular-clock"></a>
</h2>
<p>Now, we are going to look at how confident we are about our original
rate estimate, and what the implications of this is. More specifically,
we are going to generate a profile plot of the “log-transformed
probability” (in this case, this is the same as the <em>log
likelihood</em>, a term you might have heard before) as a function of
the substitution rate parameter.</p>
<p>To get the probability of observing a specific number of mutations,
you can use the <code><a href="https://rdrr.io/r/stats/Poisson.html" class="external-link">dpois()</a></code> function:</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># For instance, what is the probability of observing 10 substitutions in a 37-long </span></span>
<span><span class="co"># AA sequence after 10 million years of divergence</span></span>
<span><span class="co"># if the substitution rate is equal to 0.1 substitutions</span></span>
<span><span class="co"># per site per million years?</span></span>
<span></span>
<span><span class="va">rate_A</span> <span class="op">&lt;-</span> <span class="fl">0.1</span></span>
<span><span class="va">AA_seq_length</span> <span class="op">&lt;-</span> <span class="fl">37</span></span>
<span></span>
<span><span class="va">obs_subs</span> <span class="op">&lt;-</span> <span class="fl">10</span>  <span class="co"># amount of substitutions that have occurred</span></span>
<span><span class="va">obs_time</span> <span class="op">&lt;-</span> <span class="fl">10</span>  <span class="co"># amount of time that has passed</span></span>
<span></span>
<span><span class="va">prob_given_rate_A</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/Poisson.html" class="external-link">dpois</a></span><span class="op">(</span></span>
<span>  x <span class="op">=</span> <span class="va">obs_subs</span>, lambda <span class="op">=</span> <span class="va">obs_time</span> <span class="op">*</span> <span class="va">rate_A</span> <span class="op">*</span> <span class="va">AA_seq_length</span><span class="op">)</span></span>
<span></span>
<span><span class="va">prob_given_rate_A</span> <span class="co"># this is the likelihood</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 1.130729e-07</span></span></code></pre>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># finally, we take the (natural) log, to get</span></span>
<span><span class="co"># the log-likelihood of observing the data</span></span>
<span><span class="co"># given the model and the rate value:</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log</a></span><span class="op">(</span><span class="va">prob_given_rate_A</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] -15.99523</span></span></code></pre>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># the questions is, then: what is the likelihood of observing &gt;the same&lt; data </span></span>
<span><span class="co"># under different values of substitution rate?</span></span></code></pre></div>
<div class="section level3">
<h3 id="brief-introduction-to-intervals-derived-from-log-likelihood">Brief introduction to intervals derived from log-likelihood<a class="anchor" aria-label="anchor" href="#brief-introduction-to-intervals-derived-from-log-likelihood"></a>
</h3>
<p>When we are estimating only a single parameter, likelihood theory
shows that any likelihood value of a parameter
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>θ</mi><annotation encoding="application/x-tex">\theta</annotation></semantics></math>
given data
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>x</mi><annotation encoding="application/x-tex">x</annotation></semantics></math>
(i.e. <math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>L</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>θ</mi><mo stretchy="false" form="prefix">|</mo><mi>x</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">L(\theta | x)</annotation></semantics></math>)
that satisfies the following inequality:</p>
<p><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>L</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>θ</mi><mo stretchy="false" form="prefix">|</mo><mi>x</mi><mo stretchy="true" form="postfix">)</mo></mrow><mo>≥</mo><mrow><mo stretchy="true" form="prefix">(</mo><mi>M</mi><mi>A</mi><mi>X</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>L</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>θ</mi><mo stretchy="false" form="prefix">|</mo><mi>x</mi><mo stretchy="true" form="postfix">)</mo></mrow><mo stretchy="true" form="postfix">)</mo></mrow><mo>−</mo><mn>1.92</mn><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">\begin{equation}
L(\theta | x) \geq ( MAX(L(\theta | x)) - 1.92)
\end{equation}</annotation></semantics></math></p>
<p>will provide an estimate of the 95% confidence interval for our
parameter value.</p>
<p><br></p>
<hr>
</div>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<p>Benson, D. A., Cavanaugh, M., Clark, K., Karsch-Mizrachi, I., Lipman,
D. J., Ostell, J., &amp; Sayers, E. W. (2012). GenBank. Nucleic acids
research, 41(D1), D36-D42.</p>
<p>Hedges, S. B., &amp; Kumar, S. (2003). Genomic clocks and
evolutionary timescales. TRENDS in Genetics, 19(4), 200-206.</p>
<p>Sievers, F., Wilm, A., Dineen, D., Gibson, T. J., Karplus, K., Li,
W., … &amp; Higgins, D. G. (2011). Fast, scalable generation of
high‐quality protein multiple sequence alignments using Clustal Omega.
Molecular systems biology, 7(1), 539.</p>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Matheus Januario, Jennifer Auler, Andressa Viol, Daniel Rabosky.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.2.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
