<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Allele frequency change and selection • evolved</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Allele frequency change and selection">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">evolved</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">1.0.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../index.html"><span class="fa fa-home"></span> Home</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html"><span class="fa fa-book"></span> Reference</a></li>
<li class="active nav-item"><a class="nav-link" href="../articles/index.html"><span class="fa fa-book-open"></span> Vignettes &amp; Tutorials</a></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html"><span class="fa fa-newspaper"></span> News</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/mjanuario/evolved" aria-label="GitHub repository"><span class="fa fa-github"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Allele frequency change and selection</h1>
                        <h4 data-toc-skip class="author">Matheus
Januario, Andressa Viol, and Daniel Rabosky</h4>
            
            <h4 data-toc-skip class="date">Jan 2024</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/mjanuario/evolved/blob/v1.0.23/vignettes/popgen_selection.Rmd" class="external-link"><code>vignettes/popgen_selection.Rmd</code></a></small>
      <div class="d-none name"><code>popgen_selection.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="learning-objectives">Learning objectives<a class="anchor" aria-label="anchor" href="#learning-objectives"></a>
</h2>
<ol style="list-style-type: decimal">
<li>Breeding effective population size</li>
<li>Mutation-drift equilibrium</li>
<li>Selection</li>
<li>The case of the peppered moths</li>
</ol>
<p>First, we load our packages:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/mjanuario/evolved" class="external-link">evolved</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>We already saw in other popgen vignettes that due to sampling only,
allele frequencies are predicted to change with time in a finite
population – a process we call genetic drift. Still, many things
<em>other than drift</em> also change allele frequency. We will work
with them today.</p>
</div>
<div class="section level2">
<h2 id="breeding-effective-population-size">Breeding effective population size<a class="anchor" aria-label="anchor" href="#breeding-effective-population-size"></a>
</h2>
<p>Following Wright (1931), the breeding effective size is calculated
as:</p>
<p><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>N</mi><mi>e</mi></msub><mo>=</mo><mfrac><mrow><mn>4</mn><msub><mi>N</mi><mi>m</mi></msub><msub><mi>N</mi><mi>f</mi></msub></mrow><mrow><msub><mi>N</mi><mi>m</mi></msub><mo>+</mo><msub><mi>N</mi><mi>f</mi></msub></mrow></mfrac></mrow><annotation encoding="application/x-tex">\begin{equation}
N_e = \frac{4N_m N_f}{N_m+N_f}
\end{equation}</annotation></semantics></math></p>
</div>
<div class="section level2">
<h2 id="mutation-drift-equilibrium">Mutation-drift equilibrium<a class="anchor" aria-label="anchor" href="#mutation-drift-equilibrium"></a>
</h2>

</div>
<div class="section level2">
<h2 id="selection">Selection<a class="anchor" aria-label="anchor" href="#selection"></a>
</h2>
<p>Using the function <code><a href="../reference/NatSelSim.html">NatSelSim()</a></code> (from
<code>evolved</code>) you can explore how selection occurs in a
bi-allelic gene of a diploid organism, given a set of arguments:
<code>p0</code> is the initial frequency of the focal allele (labeled
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>A</mi><mn>1</mn></msub><annotation encoding="application/x-tex">A_1</annotation></semantics></math>
in the output plots), <code>w11</code> is the fitness of the homozygote
for the focal allele, and <code>w12</code> is the heterozygote fitness.
The function also requires the number of generations simulated
(<code>n.gen</code>).</p>
<p>The question above describes a situation where the dominant allele
has higher fitness. But there are many other possibilities. Tweak the
parameters to explore how selection behaves in different situations.</p>
<p>Now, simulate all those biological scenarios. As we said above,
before running the code, try to come up with expectations about how the
system will behave – this is fundamental for you to build intuition on
how selection works. Try to also vary the initial frequency – why is
this important?</p>
<p>Do your results correspond to your hypotheses?</p>
</div>
<div class="section level2">
<h2 id="the-case-of-the-peppered-moths">The case of the peppered moths<a class="anchor" aria-label="anchor" href="#the-case-of-the-peppered-moths"></a>
</h2>
<p>For a long time, melanic individuals of <em>Biston betularia</em>
(the peppered moth) were considered to be rare. The first dark morph was
deposited in a museum only in 1811 (Berry, 1990), 53 years after the
species was first described. The first live specimen was caught 37 years
after that by R.S. Edleston in 1848. Only 16 years later, however,
Edleston would say that the dark (a.k.a. <em>carbonaria</em>) morph was
the most common morph that he caught in his garden (Berry, 1990).</p>
<p>At the time many hypotheses were drawn to explain this phenomenon
(Cook &amp; Turner, 2020), but a very good hypothesis is that
light-colored morphs were able to camouflage easier in lichenous tree
barks (see figure 1), while the dark morph was easily seen by birds on
this background. However, after the industrial revolution, many of the
lichen died, exposing the darker tree bark underneath and supposedly
changing which morph was most visible to birds.</p>
<div class="figure">
<img src="2_morphs.png" class="r-plt" alt="Light and carbonaria morphs of the peppered moth on the same lichenous tree. Image from wikicommons." width="500"><p class="caption">
Light and carbonaria morphs of the peppered moth on the same lichenous
tree. Image from wikicommons.
</p>
</div>
<p>Almost a century ago, J. B. S. Haldane performed some calculations on
peppered moth morphs to estimate the selection coefficient
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>s</mi><annotation encoding="application/x-tex">s</annotation></semantics></math>)
that natural populations face (Haldane, 1924). Below, you will use a
similar approach to also estimate
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>s</mi><annotation encoding="application/x-tex">s</annotation></semantics></math>,
departing from the same data and assumptions that Haldane used.</p>
<p>Haldane made his estimate using analytic calculations, but we can
approach his results by using a procedure similar to the one below:</p>
<ol style="list-style-type: upper-roman">
<li>Multiply the frequency of each genotype in the population by its
relative fitness, obtaining the frequency of genotypes that will
actually reproduce.</li>
<li>Get the new allele frequency <em>after</em> selection. Then, assume
those genotypes mate randomly, which will lead to offspring genotypes
that follow frequencies expected by HWE and the new allelic frequency
(the one after selection).</li>
<li>Repeat steps I-II until selection for that characteristic is over
(i.e.,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>s</mi><mo>=</mo><mn>0</mn></mrow><annotation encoding="application/x-tex">s = 0</annotation></semantics></math>)
or there is no genetic variation anymore. Note that this is no different
than what the function <code><a href="../reference/NatSelSim.html">NatSelSim()</a></code> does “under its hood”
(which you can check by typing the function name without the parenthesis
in the R console).</li>
</ol>
<p>We need some data to help us calculate our estimates. Drawing from
the literature, Haldane assumed that the <strong>carbonaria morph alelle
had a frequency of roughly 1% in 1848 but that its frequency had
increased to 99% after 50 generations</strong>.</p>
<p>Taking Haldane’s assumptions for granted, and using the procedure
explained above, we are going to estimate the intensity of selection
favoring the dark morph. To do that:</p>
<p><strong>(1)</strong> Assume that the <em>carbonaria</em> allele is
dominant, and begin with the allele frequency given above for the
<em>carbonaria</em> allele before the industrial revolution.</p>
<p><strong>(2)</strong> Then, use step I-III above to compute the final
frequency of the allele after 50 generations of selection. After each
simulation, your value of
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>s</mi><annotation encoding="application/x-tex">s</annotation></semantics></math>
will be fixed.</p>
<p><strong>(3)</strong> Do many simulations across a range of relative
fitness values for the light morph allele.</p>
<p><strong>(4)</strong> Using the final
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>p</mi><annotation encoding="application/x-tex">p</annotation></semantics></math>
frequencies from all simulations, you will generate a “profile plot”,
where your x-axis should be the relative fitness of light morph, and
your y-axis should be the final frequency of the carbonaria allele after
50 generations.</p>
<p>Below, we designed some questions that will guide you through this
procedure.</p>
<p>Since Haldane did his approximation, subsequent research has greatly
improved our knowledge of peppered moth evolution. The fact that we have
few good temporal records from the period in which melanic individuals
were increasing in frequency obscures many details, but several
possibilities were considered for the observed change, from linkage
disequilibrium of melanic alleles to heterozygote advantage (Cook &amp;
Turner, 2020). Still, Haldane’s point is very clear: <strong>selection
can be really strong in natural populations!</strong> However, even
though the melanic morph had very high frequency at the early 20th
century, this allele was <em>not</em> fixed. Influx of alleles via
mating with migrants from elsewhere were not considered a good
hypothesis at the time but, much later, DNA-based evidence in favor of
that hypothesis was found (Cook &amp; Turner, 2020). Are you surprised
that natural populations, even in such “simple” scenarios, are
<em>that</em> complex?</p>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<p>Berry, R. J. (1990). Industrial melanism and peppered moths (Biston
betularia (L.)). Biological Journal of the Linnean Society, 39(4),
301-322.</p>
<p>Cook, L. M., &amp; Turner, J. R. (2020). Fifty per cent and all that:
what Haldane actually said. Biological Journal of the Linnean Society,
129(3), 765-771.</p>
<p>Edleston, R. S., 1864. Amptydasis betularia. The Entomologist, 2:
150</p>
<p>Haldane JBS. 1924. A mathematical theory of natural and artificial
selection. Transactions of the Cambridge Philosophical Society 23:
19–41.</p>
<p>Wright, S. (1931). Evolution in Mendelian populations. Genetics,
16(2), 97.</p>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Matheus Januario, Jennifer Auler, Andressa Viol, Daniel Rabosky.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.2.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
