<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Clocks and Rocks • evolved</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Clocks and Rocks">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">evolved</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">1.0.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../index.html"><span class="fa fa-home"></span> Home</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html"><span class="fa fa-book"></span> Reference</a></li>
<li class="active nav-item"><a class="nav-link" href="../articles/index.html"><span class="fa fa-book-open"></span> Vignettes &amp; Tutorials</a></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html"><span class="fa fa-newspaper"></span> News</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/mjanuario/evolved" aria-label="GitHub repository"><span class="fa fa-github"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Clocks and Rocks</h1>
                        <h4 data-toc-skip class="author">Matheus
Januario and Jennifer Auler</h4>
            
            <h4 data-toc-skip class="date">Dec/2023</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/mjanuario/evolved/blob/v1.0.12/vignettes/rocks_and_clocks.Rmd" class="external-link"><code>vignettes/rocks_and_clocks.Rmd</code></a></small>
      <div class="d-none name"><code>rocks_and_clocks.Rmd</code></div>
    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/mjanuario/evolved" class="external-link">evolved</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Let's also store our par() configs so </span></span>
<span><span class="co"># we can restore them whenever we change it in this tutorial</span></span>
<span><span class="va">oldpar</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span>no.readonly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>  </span></code></pre></div>
<div class="section level2">
<h2 id="the-fossil-record">The fossil record:<a class="anchor" aria-label="anchor" href="#the-fossil-record"></a>
</h2>
<p>Here we will use data on fossilized organism, which we call “fossil
occurrences”. the data was originally obtained from the Paleobiology
database, a free-to-use resource that is the standard repository for
such data and which has been used by hundreds or thousands of scientific
research articles to study the history and dynamics of biodiversity in
Deep Time.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"dinos_fossil"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">dinos_fossil</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##     phylum        class              order              family</span></span>
<span><span class="co">## 1 Chordata Ornithischia NO_ORDER_SPECIFIED    Chaoyangsauridae</span></span>
<span><span class="co">## 2 Chordata   Saurischia       Avetheropoda NO_FAMILY_SPECIFIED</span></span>
<span><span class="co">## 3 Chordata   Saurischia       Avetheropoda NO_FAMILY_SPECIFIED</span></span>
<span><span class="co">## 4 Chordata   Saurischia       Avetheropoda     Tyrannosauridae</span></span>
<span><span class="co">## 5 Chordata   Saurischia       Avetheropoda     Tyrannosauridae</span></span>
<span><span class="co">## 6 Chordata Ornithischia NO_ORDER_SPECIFIED        Ceratopsidae</span></span>
<span><span class="co">##               genus                   species early_interval late_interval</span></span>
<span><span class="co">## 1    Chaoyangsaurus     Chaoyangsaurus youngi Late Tithonian   Valanginian</span></span>
<span><span class="co">## 2 Protarchaeopteryx Protarchaeopteryx robusta Late Barremian  Early Aptian</span></span>
<span><span class="co">## 3       Caudipteryx          Caudipteryx zoui Late Barremian  Early Aptian</span></span>
<span><span class="co">## 4       Gorgosaurus      Gorgosaurus libratus Late Campanian              </span></span>
<span><span class="co">## 5       Gorgosaurus      Gorgosaurus libratus Late Campanian              </span></span>
<span><span class="co">## 6      Centrosaurus      Centrosaurus apertus Late Campanian              </span></span>
<span><span class="co">##   max_ma min_ma midpoint       lng      lat</span></span>
<span><span class="co">## 1  150.8 132.90   141.85  123.9667 42.93330</span></span>
<span><span class="co">## 2  130.0 122.46   126.23  120.7333 41.80000</span></span>
<span><span class="co">## 3  130.0 122.46   126.23  120.7333 41.80000</span></span>
<span><span class="co">## 4   83.5  70.60    77.05 -111.5287 50.74073</span></span>
<span><span class="co">## 5   83.5  70.60    77.05 -111.5493 50.73701</span></span>
<span><span class="co">## 6   83.5  70.60    77.05 -111.5289 50.73730</span></span></code></pre>
<p>Students can use functions in the package to calculate the diversity
of a certain taxonomic rank, and plot it through time.</p>
<p>To make it more fun, we will compare tow different taxonomic levels,
and will plot them in a relative, log scale:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">spDTT</span> <span class="op">=</span> <span class="fu"><a href="../reference/calcFossilDivTT.html">calcFossilDivTT</a></span><span class="op">(</span><span class="va">dinos_fossil</span>, tax.lvl <span class="op">=</span> <span class="st">"species"</span><span class="op">)</span></span>
<span><span class="va">genusDTT</span> <span class="op">=</span> <span class="fu"><a href="../reference/calcFossilDivTT.html">calcFossilDivTT</a></span><span class="op">(</span><span class="va">dinos_fossil</span>, tax.lvl <span class="op">=</span> <span class="st">"genus"</span><span class="op">)</span></span>
<span><span class="va">famDTT</span> <span class="op">=</span> <span class="fu"><a href="../reference/calcFossilDivTT.html">calcFossilDivTT</a></span><span class="op">(</span><span class="va">dinos_fossil</span>, tax.lvl <span class="op">=</span> <span class="st">"family"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># And to allow comparisons, we will use relative richness:</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">genusDTT</span><span class="op">$</span><span class="va">age</span>, xlim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rev.html" class="external-link">rev</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/range.html" class="external-link">range</a></span><span class="op">(</span><span class="va">genusDTT</span><span class="op">$</span><span class="va">age</span><span class="op">)</span><span class="op">)</span>,</span>
<span>     y<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log</a></span><span class="op">(</span><span class="va">genusDTT</span><span class="op">$</span><span class="va">div</span><span class="op">)</span><span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">max</a></span><span class="op">(</span><span class="va">genusDTT</span><span class="op">$</span><span class="va">div</span><span class="op">)</span><span class="op">)</span>, </span>
<span>     xlab<span class="op">=</span><span class="st">"Time (Million years ago)"</span>,</span>
<span>     ylab<span class="op">=</span><span class="st">"Log relative diversity"</span>,</span>
<span>     type<span class="op">=</span><span class="st">"l"</span>, col<span class="op">=</span><span class="st">"blue"</span>, ylim<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">7</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/lines.html" class="external-link">lines</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">famDTT</span><span class="op">$</span><span class="va">age</span>,</span>
<span>     y<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log</a></span><span class="op">(</span><span class="va">famDTT</span><span class="op">$</span><span class="va">div</span><span class="op">)</span><span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">max</a></span><span class="op">(</span><span class="va">famDTT</span><span class="op">$</span><span class="va">div</span><span class="op">)</span><span class="op">)</span>, </span>
<span>     col<span class="op">=</span><span class="st">"red"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/lines.html" class="external-link">lines</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">spDTT</span><span class="op">$</span><span class="va">age</span>,</span>
<span>     y<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log</a></span><span class="op">(</span><span class="va">spDTT</span><span class="op">$</span><span class="va">div</span><span class="op">)</span><span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">max</a></span><span class="op">(</span><span class="va">spDTT</span><span class="op">$</span><span class="va">div</span><span class="op">)</span><span class="op">)</span>, </span>
<span>     col<span class="op">=</span><span class="st">"black"</span><span class="op">)</span></span></code></pre></div>
<p><img src="rocks_and_clocks_files/figure-html/unnamed-chunk-4-1.png" class="r-plt" alt="" width="576" style="display: block; margin: auto;"></p>
<p>We can also visualize fossil records by running:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Family-level:</span></span>
<span><span class="fu"><a href="../reference/plotRawFossilOccs.html">plotRawFossilOccs</a></span><span class="op">(</span><span class="va">dinos_fossil</span>, tax.lvl <span class="op">=</span> <span class="st">"family"</span>, knitr <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="rocks_and_clocks_files/figure-html/unnamed-chunk-5-1.png" class="r-plt" alt="" width="576" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Genus level:</span></span>
<span><span class="fu"><a href="../reference/plotRawFossilOccs.html">plotRawFossilOccs</a></span><span class="op">(</span><span class="va">dinos_fossil</span>, tax.lvl <span class="op">=</span> <span class="st">"genus"</span>, knitr <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="rocks_and_clocks_files/figure-html/unnamed-chunk-5-2.png" class="r-plt" alt="" width="576" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Species level:</span></span>
<span><span class="fu"><a href="../reference/plotRawFossilOccs.html">plotRawFossilOccs</a></span><span class="op">(</span><span class="va">dinos_fossil</span>, tax.lvl <span class="op">=</span> <span class="st">"species"</span>, knitr <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="rocks_and_clocks_files/figure-html/unnamed-chunk-5-3.png" class="r-plt" alt="" width="576" style="display: block; margin: auto;"></p>
<p>Are they different? how could we test? Hint: look at the column names
of the fossil object:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">dinos_fossil</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##  [1] "phylum"         "class"          "order"          "family"        </span></span>
<span><span class="co">##  [5] "genus"          "species"        "early_interval" "late_interval" </span></span>
<span><span class="co">##  [9] "max_ma"         "min_ma"         "midpoint"       "lng"           </span></span>
<span><span class="co">## [13] "lat"</span></span></code></pre>
<p>Now, how complete is the record? We can use another dataset to
explored this:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"birds_spp"</span><span class="op">)</span></span></code></pre></div>
<p>And see the proportion of living species with a fossil
occurrence.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sum_w_fossil</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">birds_spp</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">dinos_fossil</span><span class="op">$</span><span class="va">species</span><span class="op">)</span></span>
<span></span>
<span><span class="va">sum_w_fossil</span> <span class="op">/</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">birds_spp</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 0.1139798</span></span></code></pre>
<p>Students can for instance explore how this varies across different
mammal groups, and which type of factors (e.g. biological, geological
factors) seem to be influencing this the most.</p>
<p>Results from this dataset can also be compared (when relevant) with
other fossil records, like those one can download from the Paleobiology
Database.</p>
<p>We can also compare the temporal trends of species number, for
instance. To help in that way, we also provide biodiversity timeseries
for more clades:</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"timeseries_fossil"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">timeseries_fossil</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##   clade    source stem_age rel_time time_ma richness</span></span>
<span><span class="co">## 1  anth Alroy2010      617  617.000   0.000     7138</span></span>
<span><span class="co">## 2  anth Alroy2010      617  609.754   7.246     7246</span></span>
<span><span class="co">## 3  anth Alroy2010      617  601.030  15.970    12673</span></span>
<span><span class="co">## 4  anth Alroy2010      617  593.970  23.030     8485</span></span>
<span><span class="co">## 5  anth Alroy2010      617  583.100  33.900    14475</span></span>
<span><span class="co">## 6  anth Alroy2010      617  575.800  41.200     9280</span></span></code></pre>
<p>It contains many fossil datasets, and we will only plot the first 4
of them</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">clades</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">timeseries_fossil</span><span class="op">$</span><span class="va">clade</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span></span>
<span></span>
<span><span class="va">cols</span><span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"#ffd353"</span>, <span class="st">"#ef8737"</span>, <span class="st">"#bb292c"</span>, <span class="st">"#62205f"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span>mfrow<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2</span>,<span class="fl">2</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="kw">for</span><span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">clades</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">aux</span> <span class="op">=</span> <span class="va">timeseries_fossil</span><span class="op">[</span><span class="va">timeseries_fossil</span><span class="op">$</span><span class="va">clade</span><span class="op">==</span><span class="va">clades</span><span class="op">[</span><span class="va">i</span><span class="op">]</span>, <span class="op">]</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">aux</span><span class="op">$</span><span class="va">time_ma</span>, <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log</a></span><span class="op">(</span><span class="va">aux</span><span class="op">$</span><span class="va">richness</span><span class="op">)</span>, col<span class="op">=</span><span class="va">cols</span><span class="op">[</span><span class="va">i</span><span class="op">]</span>, lwd<span class="op">=</span><span class="fl">3</span>,</span>
<span>       main<span class="op">=</span><span class="va">clades</span><span class="op">[</span><span class="va">i</span><span class="op">]</span>, type<span class="op">=</span><span class="st">"l"</span>, frame.plot <span class="op">=</span> <span class="cn">F</span>,</span>
<span>       xlab<span class="op">=</span><span class="st">"Time (Mya)"</span>, ylab<span class="op">=</span><span class="st">"Log richness"</span>,</span>
<span>       xlim<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/rev.html" class="external-link">rev</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/range.html" class="external-link">range</a></span><span class="op">(</span><span class="va">aux</span><span class="op">$</span><span class="va">time_ma</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
<p><img src="rocks_and_clocks_files/figure-html/unnamed-chunk-10-1.png" class="r-plt" alt="" width="576" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Restoring old par() configs:</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span><span class="va">oldpar</span><span class="op">)</span></span></code></pre></div>
<p>Students can explore this other dataset, and compare it with the
previously discussed ones. They can compare richness thought time,
calculate statistics related to richness change. Explore factors
(e.g. mass extinction) that might have affected some clades, among other
learning problems.</p>
<p>But fossils are not the only way to explore the timescale of
evolution. Below, we are going to show how some functions that use this
type of data work.</p>
</div>
<div class="section level2">
<h2 id="comparing-molecular-patterns">Comparing molecular patterns<a class="anchor" aria-label="anchor" href="#comparing-molecular-patterns"></a>
</h2>
<p>With other functions, students can also explore molecular sequences
and compare species.</p>
<p>First we load the dataset of protein sequences from the cytochrome
oxidase 1 gene. This gene, often known as <code>CO1</code>, is a
mitochondrial gene that plays a key role in cellular respiration (e.g.,
the primary aerobic pathway to energy ( ATP ) generation).
<code>CO1</code> contains approximately 513 aminoacids (AA) and has been
used by previous studies for reconstructing phylogenetic trees and
estimating divergence times between taxa by assuming a molecular
clock:</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">cytOxidase</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">cytOxidase</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## 17 amino acid sequences, each with length 513</span></span></code></pre>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">cytOxidase</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## $cnidaria</span></span>
<span><span class="co">## [1] "-RWIFSTNHKDIGTLYL"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $snake</span></span>
<span><span class="co">## [1] "TRWLFSTNHKDIGTLYL"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $echinoderm</span></span>
<span><span class="co">## [1] "NRWLFSTNHKDIGTLYL"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $mollusk</span></span>
<span><span class="co">## [1] "MRWLFSTNHKDIGTLYI"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $alligator</span></span>
<span><span class="co">## [1] "HRWFFSTNHKDIGTLYF"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $lamprey</span></span>
<span><span class="co">## [1] "IRWLFSTNHKDIGTLYL"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $shark</span></span>
<span><span class="co">## [1] "NRWLFSTNHKDIGTLYL"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $bird</span></span>
<span><span class="co">## [1] "NRWLFSTNHKDIGTLYL"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $frog</span></span>
<span><span class="co">## [1] "TRWLFSTNHKDIGTLYL"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $fish</span></span>
<span><span class="co">## [1] "TRWLFSTNHKDIGTLYL"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $platypus</span></span>
<span><span class="co">## [1] "NRWLFSTNHKDIGTLYL"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $human</span></span>
<span><span class="co">## [1] "DRWLFSTNHKDIGTLYL"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $chimpanzee</span></span>
<span><span class="co">## [1] "DRWLFSTNHKDIGTLYL"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $bryozoa</span></span>
<span><span class="co">## [1] "MRWLGSTNHKDIGTLYF"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $annelid</span></span>
<span><span class="co">## [1] "MRWLYSTNHKDIGTLYF"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $insect</span></span>
<span><span class="co">## [1] "RQWLFSTNHKDIGTLYF"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $crustacea</span></span>
<span><span class="co">## [1] "RQWLFSTNHKDIGTLYL"</span></span></code></pre>
<p>We can compare two sequences in terms AA difference number. For
instance if we want to compare a species of snake with one species of
bird, we type:</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/countSeqDiffs.html">countSeqDiffs</a></span><span class="op">(</span><span class="va">cytOxidase</span>, <span class="st">"snake"</span>, <span class="st">"bird"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 123</span></span></code></pre>
<p>And to calculate the proportion of differences, we type:</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/countSeqDiffs.html">countSeqDiffs</a></span><span class="op">(</span><span class="va">cytOxidase</span>, <span class="st">"snake"</span>, <span class="st">"bird"</span><span class="op">)</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/nchar.html" class="external-link">nchar</a></span><span class="op">(</span><span class="va">cytOxidase</span><span class="op">[</span><span class="st">"snake"</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##     snake </span></span>
<span><span class="co">## 0.2397661</span></span></code></pre>
<p>We can also quickly visualize the sequences by directly handling all
elements in this object:</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotProteinSeq.html">plotProteinSeq</a></span><span class="op">(</span><span class="va">cytOxidase</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"snake"</span>, <span class="st">"bird"</span>, <span class="st">"cnidaria"</span><span class="op">)</span>, knitr <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="rocks_and_clocks_files/figure-html/unnamed-chunk-14-1.png" class="r-plt" alt="" width="576" style="display: block; margin: auto;"></p>
<p>And using patterns of molecular divergence, as well as fossil
occurrences, we can build and, specially, date, molecular
phylogenies.</p>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Matheus Januario, Jennifer Auler, Andressa Viol, Daniel Rabosky.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.2.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
